collaboration/sig-core/proposals/event-flow.md

```diff
@@ -0,0 +1,224 @@
+- Consequence could be a user namespace without Knative Broker => **No eventing!**
+- Therefore, deprovisioning of a Knative Broker is **not implemented** at the moment.
+- (*) The only way I can think of solving this with the current architecture is having a **mutex** in provisioner/deprovisioner:
+    - In the provisioning step we need to check that the Knative Broker exists. If not, label the namespace and wait for successful creation of the Broker.
+    - In the deprovisioning step, we need a way to check if the Knative Broker is in use, e.g. listing all successfully provisioned Service Instances belonging to Application-Broker.
+    - If the Broker is not in use, we can delete it. No provisioning can happen meanwhile (mutex), therefore no race.
+
+3: [Reflect Status of Knative Broker on ServiceInstance provisioning status #7696](https://github.com/kyma-project/kyma/issues/7696)
+
+TL;DR:
+- We can check the status of the Broker, but it would block the provisioning request for some time. If the broker is not ready, we mark the provisioning request as failed. However, if the broker heals itself (e.g. reconciliation or dependent objects heal), 
+  the status of the Service Instance will stay in failed status.
+
+## Possible Solutions
+
+### Custom Eventing Controller
+
+Instead of performing steps 2-4 inside Application Broker, the necessary steps to enable eventing for a Kyma Application should be `delegated` to a `new Kubernetes controller`.
+
+How would we solve the above mentioned problems with this solution:
+
+*Problem 1*: The `EventFlow` CR (Custom Resource) would be created by the Application Broker.
+   The chance that the creation of EventFlow CR fails due to an error in the provisioning request (inside Application Broker - no retries) is very low.
+   Therefore, retries in the Application Broker are not required anymore.
+   If the Application Channel is not ready, an Informer on the Knative Service (which reflects the status of the Application Channel as well) would trigger a new reconciliation,
+   therefore implementing the required retries.
+
+*Problem 2*: Whenever a Service Instance is provisioned by the Application Broker, an EventFlow CR would be created in the same namespace as well.
+   That means we know which applications are relying on a Knative Broker.
+   When a deprovisiong request arrives in the Application Broker, we delete the appropriate EventFlow CR.
+   If there are no EventFlow CRs left, we can safely delete the Knative Broker.
+   If a new provisioning request arrives in the meantime, we wait for the deprovisioning of the Knative Broker and trigger the creation of a new Knative Broker. If the broker is ready, the EventFlow CR status is set to ready.
+
+*Problem 3*: We can easily reflect the status of the Knative Broker inside the EventFlow status using an Informer on the Knative Broker. We won't reflect the status of the Knative Broker in the Service Instance.
```

Comment:
When `EventFlow` is not ready and provisioning of `ServiceInstance` is good, then also eventing is not enabled in the namespace. Is there a way to mark the provisioning of `ServiceInstance` as failed? I think this is important because when provisioning is good, it sends a message to the user as eventing is enabled. And this is not true. It all depends on the status of EventFlow. So I think it needed to be reflected.

Reply:
Totally agree. @k15r had the same argument. Let's move discussion over to https://github.com/kyma-project/community/pull/443#discussion_r398369134.

Reply:
I updated the proposal with the necessary changes required to the UI. Please check again :)